c_glib/arrow-glib/array-builder.h

```diff
@@ -947,46 +550,16 @@ gboolean garrow_binary_array_builder_append_null(GArrowBinaryArrayBuilder *build
 
 #define GARROW_TYPE_STRING_ARRAY_BUILDER        \
   (garrow_string_array_builder_get_type())
-#define GARROW_STRING_ARRAY_BUILDER(obj)                        \
-  (G_TYPE_CHECK_INSTANCE_CAST((obj),                            \
-                              GARROW_TYPE_STRING_ARRAY_BUILDER, \
-                              GArrowStringArrayBuilder))
-#define GARROW_STRING_ARRAY_BUILDER_CLASS(klass)                \
-  (G_TYPE_CHECK_CLASS_CAST((klass),                             \
-                           GARROW_TYPE_STRING_ARRAY_BUILDER,    \
-                           GArrowStringArrayBuilderClass))
-#define GARROW_IS_STRING_ARRAY_BUILDER(obj)                             \
-  (G_TYPE_CHECK_INSTANCE_TYPE((obj),                                    \
-                              GARROW_TYPE_STRING_ARRAY_BUILDER))
-#define GARROW_IS_STRING_ARRAY_BUILDER_CLASS(klass)             \
-  (G_TYPE_CHECK_CLASS_TYPE((klass),                             \
-                           GARROW_TYPE_STRING_ARRAY_BUILDER))
-#define GARROW_STRING_ARRAY_BUILDER_GET_CLASS(obj)              \
-  (G_TYPE_INSTANCE_GET_CLASS((obj),                             \
-                             GARROW_TYPE_STRING_ARRAY_BUILDER,  \
-                             GArrowStringArrayBuilderClass))
-
-typedef struct _GArrowStringArrayBuilder         GArrowStringArrayBuilder;
-typedef struct _GArrowStringArrayBuilderClass    GArrowStringArrayBuilderClass;
-
-/**
- * GArrowStringArrayBuilder:
- *
- * It wraps `arrow::StringBuilder`.
- */
-struct _GArrowStringArrayBuilder
-{
-  /*< private >*/
-  GArrowBinaryArrayBuilder parent_instance;
-};
-
+G_DECLARE_DERIVABLE_TYPE(GArrowStringArrayBuilder,
+                         garrow_string_array_builder,
+                         GARROW,
+                         STRING_ARRAY_BUILDER,
+                         GArrowArrayBuilder)
```

Comment:
We need to use `GArrowBinaryArrayBuilder` as the parent.

Reply:
Thank you for catching this.  I've fixed it.